spring-cloud-kubernetes-test-support/src/main/java/org/springframework/cloud/kubernetes/integration/tests/commons/Commons.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.integration.tests.commons;
+
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+import com.github.dockerjava.api.model.Image;
+import org.testcontainers.k3s.K3sContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import static org.springframework.cloud.kubernetes.integration.tests.commons.K8SUtils.getPomVersion;
+
+/**
+ * A few commons things that can be re-used across clients. This is meant to be used for
+ * testing purposes only.
+ *
+ * @author wind57
+ */
+public final class Commons {
+
+	private Commons() {
+		throw new AssertionError("No instance provided");
+	}
+
+	/**
+	 * Rancher version to use for test-containers.
+	 */
+	public static final String RANCHER = "rancher/k3s:v1.21.10-k3s1";
+
+	/**
+	 * Command to use when starting rancher. Without "server" option, traefik si not
+	 * installed
+	 */
+	public static final String RANCHER_COMMAND = "server";
+
+	/**
+	 * Test containers exposed ports.
+	 */
+	public static final int[] EXPOSED_PORTS = new int[] { 80, 6443, 8080, 8888, 9092 };
+
+	/**
+	 * Temporary folder where to load images.
+	 */
+	public static final String TEMP_FOLDER = new File(System.getProperty("java.io.tmpdir")).getAbsolutePath();
+
+	private static final K3sContainer CONTAINER = new FixedPortsK3sContainer(DockerImageName.parse(Commons.RANCHER))
+			.configureFixedPorts(EXPOSED_PORTS).withFileSystemBind(TEMP_FOLDER, TEMP_FOLDER)
+			.withCommand(Commons.RANCHER_COMMAND).withReuse(true);
+
+	public static K3sContainer container() {
+		return CONTAINER;
+	}
+
+	public static void loadImage(String image) throws Exception {
```

Comment:
@wind57 can this be generalized you think?  Due to the way our internal build system works, I am thinking we need to pull all external images before we start the container and load them in so k8s is not pulling them.  So could we generalize this to load other images as well?

I am thinking we pull images like istio, rabbitmq, wiremock, zookeeper, etc and have the tests load them into the container here.  We did something similar with Kind.

Reply:
I see your point. Ill take a look, sure

Reply:
Im looking at it now....

Reply:
ok, I have some minor ideas here too, let's see if we reach common ground. I'll do a minor test soon too, if you do not mind.

Reply:
this is definitely doable. if you would let me, I will raise a PR tomorrow

Reply:
I have some code thats almost ready to go here, I will probably commit it tonight.  Unfortunately it wont be in the form of a PR since I want to test it in our Jenkins environment, but I will let you know when I commit it.  If you want to make some enhancements to it from there (or if I completely screw things up, which is likely) thats cool